enderio-machines/src/main/java/com/enderio/machines/common/blockentity/VacuumChestBlockEntity.java

```diff
@@ -0,0 +1,133 @@
+import com.enderio.machines.common.blockentity.data.sidecontrol.item.ItemHandlerMaster;
+import com.enderio.machines.common.blockentity.data.sidecontrol.item.ItemSlotLayout;
+import com.enderio.machines.common.menu.VacuumChestMenu;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.item.ItemEntity;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.AABB;
+
+public class VacuumChestBlockEntity extends MachineBlockEntity {
+    private static final double COLLISION_DISTANCE_SQ = 1 * 1;
+    private static final double SPEED = 0.025;
+    private static final double SPEED_4 = SPEED*4 ;
+    private static final int MAX_RANGE = 6;
+    private int range = 6;
+    private List<WeakReference<ItemEntity>> itemEntities = new ArrayList<>();
+    
+    public VacuumChestBlockEntity(BlockEntityType<?> pType, BlockPos pWorldPosition, BlockState pBlockState) {
+        super(MachineTier.STANDARD, pType, pWorldPosition, pBlockState);
+        add2WayDataSlot(new IntegerDataSlot(() -> this.range, this::setRange, SyncMode.GUI));
+    }
+    
+    @Override
+    public AbstractContainerMenu createMenu(int containerId, Inventory inventory, Player player) {
+        return new VacuumChestMenu(this, inventory, containerId);
+    }
+    
+    @Override
+    public Optional<ItemSlotLayout> getSlotLayout() {
+        return Optional.of(ItemSlotLayout.basic(28,0));
+    }
+    
+    @Override
+    protected ItemHandlerMaster createItemHandler(ItemSlotLayout layout) {
+        return new ItemHandlerMaster(getIoConfig(), layout) {
+            @Override
+            protected void onContentsChanged(int slot) {
+                setChanged();
+            }
+            
+            @Override
+            public ItemStack insertItem(int slot, ItemStack stack, boolean simulate) {
+                if (slot == 27) {
+                    return stack;
+                }
+                return super.insertItem(slot, stack, simulate);
+            }
+        };
+    }
+    
+    @Override
+    public void tick() {
+        if (this.getRedstoneControl().isActive(level.hasNeighborSignal(worldPosition))) {
+            this.collectItems(this.getLevel(), this.getBlockPos(), this.range);
+        }
+        super.tick();
+    }
+    
+    private void collectItems(Level level, BlockPos pos, int range) {
+        if ((level.getGameTime() + pos.asLong()) % 5 == 0) {
+            AABB area = new AABB(pos).inflate(range);
+            for (ItemEntity ie:level.getEntitiesOfClass(ItemEntity.class, area, e -> true)) {
+                this.itemEntities.add(new WeakReference<ItemEntity>(ie));
```

Comment:
```suggestion
                this.itemEntities.add(new WeakReference(ie));
```
this generic isn't necessary, also move the todo comment  to e-> true

Reply:
It may not be needed, but my ide does warn me about it. It shouldn't matter though?